src/Commands/Redis/CacheForRedis/CacheListCommand.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.CommandLine.Parsing;
+using AzureMcp.Arguments.Redis.CacheForRedis;
+using AzureMcp.Models.Command;
+using AzureMcp.Services.Interfaces;
+using Microsoft.Extensions.Logging;
+using ModelContextProtocol.Server;
+
+namespace AzureMcp.Commands.Redis.CacheForRedis;
+
+/// <summary>
+/// Lists Azure Cache for Redis resources (Basic, Standard, and Premium tier caches) in the specified subscription.
+/// </summary>
+public sealed class CacheListCommand(ILogger<CacheListCommand> logger) : SubscriptionCommand<CacheListArguments>()
+{
+    private readonly ILogger<CacheListCommand> _logger = logger;
+
+    protected override string GetCommandName() => "list";
```

Comment:
Please rebase with main.  We update the BaseCommand interface. Make sure you add the title property as well and update the McpServerTool attribute with the title.